web/models.py

```
@@ -1753,17 +1753,30 @@ def display_name(self):
-        ("challenge", "Challenge Completion"),
-        ("course", "Course Completion"),
+    """Model representing badges awarded for various achievements."""
+
+    BADGE_TYPE_CHOICES = [
+        ("perfect_attendance", "Perfect Attendance"),
+        ("participation", "Outstanding Participation"),
+        ("completion", "Course Completion"),
         ("achievement", "Special Achievement"),
-        ("teacher_awarded", "Teacher Awarded"),
+        ("challenge", "Challenge Completion"),
     ]
```

Ruff (0.8.2)

Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)